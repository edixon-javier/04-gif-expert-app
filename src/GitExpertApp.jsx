import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["Deat Note"]);

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return;
    setCategories([newCategory, ...categories]);
    //? la linea comentada, lo que dice es que hace referencia al estado local de categorias cat y despues si hace el resto.
    // setCategories(cats => [...cats, 'Naruto']);
  };

  return (
    <>
      <h1>Expert Gif </h1>
      <AddCategory onNewValue={onAddCategory} />
      {categories.map((cat) => (
        <GifGrid category={cat} key={cat} />
      ))}
    </>
  );
};
