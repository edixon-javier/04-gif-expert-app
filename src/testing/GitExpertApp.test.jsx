import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../GitExpertApp";
/* eslint-disable no-undef */
describe("test for the component GitExpertApp", () => {
  test("Render component succesfully", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
    screen.debug();
  });
  test("Render title for the component", () => {
    const { container } = render(<GifExpertApp />);
    expect(container).toMatchSnapshot();
    screen.debug();
  });
});
