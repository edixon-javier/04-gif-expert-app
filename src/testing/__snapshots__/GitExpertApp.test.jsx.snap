// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for the component GitExpertApp Render component succesfully 1`] = `
<div>
  <h1>
    Expert Gif 
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    Deat Note
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`test for the component GitExpertApp Render title for the component 1`] = `
<div>
  <h1>
    Expert Gif 
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    Deat Note
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;

exports[`test for the component GitExpertApp change value for the input value 1`] = `
<div>
  <h1>
    Expert Gif 
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs..."
      type="text"
      value=""
    />
  </form>
  <h3>
    Deat Note
  </h3>
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
